[Fix #1142] Don't sniff for ports when port is present in cider-known-endpoint

vspinu opened the pull request:
We can think of making the port detection more robust, but I think it should stay. It's robust for most common case of nREPL running locally.

bbatsov commented:
Did you get any comments from me. It seems they disappeared. Btw, don't forget to update the changelog.

vspinu commented:
> Did you get any comments from me. It seems they disappeared.

No I didn't. I fixed a couple of typos by forced push. That might have screwed the comments.

>  Btw, don't forget to update the changelog.

Added.

vspinu commented:
> No I didn't.

Actually I did. Fixing now.

vspinu commented:
Done.

bbatsov commented:
:+1:
